app/models/miq_worker.rb

```diff
@@ -267,6 +267,8 @@ def self.start_worker(*params)
     w
   end
 
+  cache_with_timeout(:my_worker) { server_scope.find_by(:pid => Process.pid) }
```

Comment:
It might be weird if we're in a priority worker process and `MiqGenericWorker.my_worker` returned a `MiqPriorityWorker` row.